cfn-resources/private-endpoint-service/cmd/resource/resource.go

```diff
@@ -0,0 +1,282 @@
+				map[string]interface{}{"stateName": "DELETING"}, currentModel, 20), nil
+		}
+	}
+	if err != nil {
+		return progressevent.GetFailedEventByResponse(fmt.Sprintf("Error getting resource : %s", err.Error()),
+			response.Response), nil
+	}
+
+	if privateEndpointResponse == nil {
+		return progressevent.GetFailedEventByCode(fmt.Sprintf("Error deleting resource, private Endpoint Response is null : %s", err.Error()),
+			cloudformation.HandlerErrorCodeNotFound), nil
+	}
+
+	response, err = mongodbClient.PrivateEndpoints.Delete(context.Background(), *currentModel.ProjectId,
+		providerName,
+		*currentModel.Id)
+
+	if err != nil {
+		return progressevent.GetFailedEventByResponse(fmt.Sprintf("Error getting resource : %s", err.Error()),
+			response.Response), nil
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus:      handler.InProgress,
+		Message:              "Delete in progress",
+		ResourceModel:        currentModel,
+		CallbackDelaySeconds: 20,
+		CallbackContext: map[string]interface{}{
+			"stateName": "DELETING",
+		}}, nil
+}
+
+// List handles the List event from the Cloudformation service.
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	if errEvent := validator.ValidateModel(ListRequiredFields, currentModel); errEvent != nil {
+		_, _ = logger.Warnf("Validation Error")
+		return *errEvent, nil
+	}
+
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+
+	mongodbClient, pe := util.NewMongoDBClient(req, currentModel.Profile)
+	if pe != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", *pe)
+		return *pe, nil
+	}
+
+	params := &mongodbatlas.ListOptions{
+		PageNum:      0,
+		ItemsPerPage: 100,
+	}
+
+	privateEndpointResponse, response, err := mongodbClient.PrivateEndpoints.List(context.Background(),
+		*currentModel.ProjectId,
+		providerName,
+		params)
+	if err != nil {
+		return progressevent.GetFailedEventByResponse(fmt.Sprintf("Error listing resource : %s", err.Error()),
+			response.Response), nil
+	}
+
+	mm := make([]interface{}, 0, len(privateEndpointResponse))
+	for i := range privateEndpointResponse {
+		var m Model
+		m.completeByConnection(privateEndpointResponse[i])
+		m.Region = currentModel.Region
+		m.Profile = currentModel.Profile
+		m.ProjectId = currentModel.ProjectId
+		mm = append(mm, m)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "List successful",
+		ResourceModels:  mm}, nil
+}
+
+func isDeleting(req handler.Request) bool {
+	callback := req.CallbackContext["stateName"]
+	if callback == nil {
+		return false
+	}
+
+	callbackValue := fmt.Sprintf("%v", callback)
+	return callbackValue == "DELETING"
+}
+
+func (m *Model) completeByConnection(c mongodbatlas.PrivateEndpointConnection) {
+	m.Id = &c.ID
+	m.EndpointServiceName = &c.EndpointServiceName
+	m.ErrorMessage = &c.ErrorMessage
+	m.Status = &c.Status
+
+	copy(m.InterfaceEndpoints, c.InterfaceEndpoints)
```

Comment:
particular reason for a copy vs a normal assign statement?
On another hand, name of `completeByConnection` is hard to understand, would favor `populatePrivateEndpointProperties`

Reply:
i remember i had some issues with the pointers, they were replaced in the test, let me check again and ill let you know